web/src/app/change-password/page.tsx

```diff
@@ -1,7 +1,57 @@
+import { revalidatePath } from "next/cache";
 import CompanyLogo from "../components/company-logo";
-import ChangePasswordForm from "./components/change-password-form";
+import ChangePasswordForm, {
+    ChangePasswordFormSchema,
+} from "./components/change-password-form";
+import { cookies } from "next/headers";
+import { redirect } from "next/navigation";
+
+function getSession() {
+    const session = cookies().get("session")?.value;
+
+    if (!session) {
+        return null;
+    }
+
+    return session;
+}
 
 export default async function ChangePasswordPage() {
+    const session = getSession();
+
+    if (!session) {
+        return redirect("/login");
+    }
+
+    async function changePassword(data: ChangePasswordFormSchema) {
+        "use server";
+
+        const body = {
+            ...data,
+            currentPassword: String(data.currentPassword),
+            newPassword: String(data.newPassword),
+            repeatPassword: String(data.repeatPassword),
+        };
+
+        const token = `Bearer ${session}`;
+
+        const response = await fetch("http://api:8080/change-password", {
+            method: "PATCH",
+            headers: {
+                "content-type": "application/json",
+                authorization: token,
+            },
+            body: JSON.stringify(body),
+        });
```

Comment:
```suggestion
      const response = await fetch("http://api:8080/change-password", {
            method: "PATCH",
            headers: {
                "content-type": "application/json",
                authorization: `Bearer ${session}`,
            },
            body: JSON.stringify(body),
        });
```

Deixa isso no header só pra ficar bem claro que tipo de autenticação é usado aqui, no caso `Bearer`.